### PR TITLE
HEEDLS-858 Add migration to snapshot database before data migration

### PR DESCRIPTION
**Pull request by AlexJacksonDS**
### JIRA link
https://softwiretech.atlassian.net/browse/HEEDLS-858

### Description
Adds a migration to snapshot the current database, saving in the default location with a timestamp name.

### Screenshots
N/A

-----
### Developer checks
(Leave tasks unticked if they haven't been appropriate for your ticket.)

I have:
- [ ] Run the formatter and made sure there are no IDE errors.
- [ ] Written tests for the changes (accessibility tests, unit tests for controller, data services, services, view models, etc)
- [x] Manually tested my work with and without JavaScript. Full manual testing guidelines can be found here: https://softwiretech.atlassian.net/wiki/spaces/HEE/pages/6703648740/Testing
- [ ] Updated/added documentation in Swiki and/or Readme. Links (if any) are below:
  - [doc_1_here](link_1_here)
- [ ] Updated my Jira ticket with information about other parts of the system that were touched as part of the MR and have to be sanity tested to ensure nothing’s broken.
- [x] Scanned over my own MR to ensure everything is as expected.

### CODE REVIEW

File: DigitalLearningSolutions.Data.Migrations/Properties/Resources.Designer.cs [hunk 1]
```diff
@@ -501,22 +501,22 @@ internal class Resources {
...
+        ///DROP TABLE dbo.FrameworkCompetenciesHistory;
+        ///GO
+        ///
+        ///-- Remove versioning from FrameworkCompetencyGroups table
+        ///ALTER TABLE FrameworkCompetencyGroups SET (SYSTEM_VERSIONING = OFF);
+        ///DROP TABLE dbo.FrameworkCompetencyGroups;
+        ///DROP TABLE dbo.FrameworkCompetencyGroupsHistory;
+        ///GO
+        ///
+        ///-- Remove versioni [rest of string was truncated]&quot;;.
```
Comment (JBsoftwire) [inline DigitalLearningSolutions.Data.Migrations/Properties/Resources.Designer.cs:519]: Is this a typo?

**Review by JBsoftwire — CHANGES_REQUESTED** [review]

File: DigitalLearningSolutions.Data.Migrations/Properties/Resources.Designer.cs (hunk 1 above)
Comment (AlexJacksonDS) [reply DigitalLearningSolutions.Data.Migrations/Properties/Resources.Designer.cs:519]: I guess there is a typo in that SQL script? this is auto generated stuff

Comment (AlexJacksonDS) [reply DigitalLearningSolutions.Data.Migrations/Properties/Resources.Designer.cs:519]: It's just truncated at that point in one of several lines in a script that look like this "-- Remove versioning from Competencies table"

Comment (JBsoftwire) [reply DigitalLearningSolutions.Data.Migrations/Properties/Resources.Designer.cs:519]: Weird, did you maybe have the window highlighted at some point? I've typed `git [whatever]` into a .cs file a dozen times. If it's autogenerated maybe we just leave it.

Comment (JBsoftwire) [reply DigitalLearningSolutions.Data.Migrations/Properties/Resources.Designer.cs:519]: That explains it, I couldn't work out what it was but of course it's 'versioni-ng'. I'll pass this then.

Comment (AlexJacksonDS) [reply DigitalLearningSolutions.Data.Migrations/Properties/Resources.Designer.cs:519]: I added a new SQL file into the resources, I guess it just decided to generate slightly differently this time

**Review by JBsoftwire — APPROVED** [review]

**Review by DanBloxham-sw — APPROVED** [review]
I think this looks good.

**Pull request merged**